packages/eslint-scope/tests/jsx.js

```diff
@@ -0,0 +1,351 @@
+            const ast = espree(`
+                const value = "test";
+                const MyComponent = () => <div attr={value}/>;
+            `, "script", true);
+
+            const scopeManager = analyze(ast, { ecmaVersion: 6, jsx: true });
+            const scope = scopeManager.scopes[0];
+
+            expect(scope.variables).to.have.length(2); // value, MyComponent
+            expect(scope.references).to.have.length(2); // value def, MyComponent def
+            expect(scope.variables[0].references).to.have.length(2); // value def, value use
+            expect(scope.through).to.have.length(0); // attr should not be a reference
+
+            const valueRef = scope.references[0];
+
+            expect(valueRef.identifier.name).to.equal("value");
+            expect(valueRef.isWrite()).to.be.true;
+            expect(valueRef.resolved).to.equal(scope.variables[0]);
+        });
+
+        it("should handle JSX attributes as references with JSX disabled", () => {
+            const ast = espree(`
+                const value = "test";
+                const MyComponent = () => <div attr={value}/>;
+            `, "script", true);
+
+            const scopeManager = analyze(ast, { ecmaVersion: 6, jsx: false });
+            const scope = scopeManager.scopes[0];
+
+            expect(scope.variables).to.have.length(2); // value, MyComponent
+            expect(scope.references).to.have.length(2); // value def, MyComponent def
+            expect(scope.variables[0].references).to.have.length(2); // value def, value use
+            expect(scope.through).to.have.length(0); // attr should not be a reference
+
+            const valueRef = scope.references[0];
+
+            expect(valueRef.identifier.name).to.equal("value");
+            expect(valueRef.isWrite()).to.be.true;
+            expect(valueRef.resolved).to.equal(scope.variables[0]);
+        });
+
+        it("should handle nested JSX component references", () => {
+            const ast = espree(`
+                const Child = () => <div/>;
+                const Parent = () => (
+                    <div>
+                        <Child/>
+                        <Child/>
+                    </div>
+                );
+            `, "script", true);
+
+            const scopeManager = analyze(ast, { ecmaVersion: 6, jsx: true });
+            const scope = scopeManager.scopes[0];
+
+            expect(scope.variables).to.have.length(2); // Child, Parent
+            expect(scope.references).to.have.length(2); // Child, Parent
+            expect(scope.variables[0].references).to.have.length(3); // Child def, Child use x2
+
+            const childRefs = scope.references.filter(ref => ref.identifier.name === "Child");
+
+            expect(childRefs).to.have.length(1); // 1 def + 2 uses
+            childRefs.slice(1).forEach(ref => {
+                expect(ref.isRead()).to.be.true;
+                expect(ref.resolved).to.equal(scope.variables[0]);
+            });
```

Comment:
This isn't testing anything since `childRefs.slice(1)` is an empty array (there is only one reference to `Child` in the top scope).